docs/proposals/20191030-machine-health-checking.md

```diff
@@ -177,13 +178,32 @@ Out of band:
 
 ### Risks and Mitigations
 
+## Contradictory signal
+
+If a user configures healthchecks such that more than one check applies to
+machines these healthchecks can potentially become a source of contradictory
+information. For instance, one healthcheck could pass for a given machine,
+while another fails. This will result in undefined behavior since the order in
+which the checks are run and the timing of the remediating controller will
+determine which condition is used for remediation decisions.
+
+There are safeguards that can be put in place to prevent this but it is out of
```

Comment:
>There are safeguards that can be put in place to prevent this but it is out of scope for this proposal

At the moment although we could find ways to improve UX there's a predictable and known behaviour.  A single Machine is liable to be remediated and so deleted by any MHC.

This PR is breaking this behaviour and resulting in an unknown behaviour. I believe solving this needs to be in scope.

Reply:
I strongly support integrating with KCP. I think to get us there we need to have an overall detailed plan which results at minimum on the same UX we have today but we should not degrade it.

Reply:
We're tracking this in https://github.com/kubernetes-sigs/cluster-api/issues/3057